pkg/remote/kyma_synchronization_context_test.go

```diff
@@ -0,0 +1,83 @@
+package remote_test
+
+import (
+	"testing"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/pkg/remote"
+	"github.com/kyma-project/lifecycle-manager/pkg/testutils"
+	"github.com/stretchr/testify/assert"
+	v1extensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+// nolint:funlen
+func TestUpdateKymaAnnotations(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		kyma              *v1beta2.Kyma
+		kcpCRD            *v1extensions.CustomResourceDefinition
+		skrCRD            *v1extensions.CustomResourceDefinition
+		kcpAnnotationName string
+		skrAnnotationName string
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "Update ModuleTemplate Annotations",
+			args: args{
+				kyma: testutils.NewTestKyma("test-kyma"),
+				skrCRD: &v1extensions.CustomResourceDefinition{
```

Comment:
1. regarding how to design a unit test

these two crd, you don't have to offer as args, you can create them in the test body.

The template will automatically create that args struct, but most time, you'd better adjust them and don't have to use the original one, the args should be easy to read, the idea is to design the input value and expected value for test, the input value is basically for your function, and expect value is the output of the function.

2. this `UpdateKymaAnnotations` function is just for updating annotation, it does not bring much value to writing a test for it, and it's not critical for this feature request.

Reply:
for point 1, you can have a look at this test as a reference: https://github.com/kyma-project/lifecycle-manager/blob/main/api/v1beta2/moduletemplate_webhook_test.go

Reply:
Yes this test now is not of value. I had it earlier because I did the check here, so it served as a test for the conditions as well.